backend/src/Service/Admin/Order/AdminOrderService.php

```diff
@@ -1954,4 +1960,127 @@ public function getCaptainsWhoDeliveredLastFiveOrders(): array
 
         return $response;
     }
+
+    public function getNewAdminCalculateCostDeliveryOrderRequestObject(): AdminCalculateCostDeliveryOrderRequest
+    {
+        return new AdminCalculateCostDeliveryOrderRequest();
+    }
+
+    /**
+     * Calculate the delivery cost according to store owner profile id and distance, and return the total value of
+     * the delivery cost
+     */
+    public function calculateDeliveryCostByStoreOwnerProfileIdAndDistance(int $storeOwnerProfileId, float $distance): float
+    {
+        $adminCalculateCostDeliveryOrderRequest = $this->getNewAdminCalculateCostDeliveryOrderRequestObject();
+
+        $adminCalculateCostDeliveryOrderRequest->setStoreOwnerProfileId($storeOwnerProfileId);
+        $adminCalculateCostDeliveryOrderRequest->setStoreBranchToClientDistance($distance);
+
+        $adminCalculateCostDeliveryOrderRequest = $this->calculateCostDeliveryOrder($adminCalculateCostDeliveryOrderRequest);
+
+        return $adminCalculateCostDeliveryOrderRequest->total;
+    }
+
+    /**
+     * Create order/s by admin for testing/debugging purposes
+     */
+    public function createTestingOrderByAdmin(TestingOrderCreateByAdminRequest $request, int $userId): CanCreateOrderResponse|array|string
+    {
+        $response = [];
+
+        if ((! $request->getOrdersCount())
+            || ($request->getOrdersCount() <= 0)
+            || ($request->getOrdersCount() > 3)) {
+            $request->setOrdersCount(1);
+        }
+
+        // 1 check if store owner profile does exist
+        $storeOwnerProfile = $this->storeOwnerProfileService->getStoreOwnerProfileEntityByIdForAdmin($request->getStoreOwner());
+
+        if ($storeOwnerProfile === null) {
+            return StoreProfileConstant::STORE_OWNER_PROFILE_NOT_EXISTS;
+        }
+
+        // 2 check if store branch exist
+        $branch = $this->storeOwnerBranchService->getBranchEntityById($request->getBranch());
+
+        if (! $branch) {
+            return StoreOwnerBranch::BRANCH_NOT_FOUND;
+        }
+
+        $request->setStoreOwner($storeOwnerProfile);
+        $request->setBranch($branch);
+
+        for ($counter = 0; $counter < $request->getOrdersCount(); $counter++) {
```

Comment:
please check if there is '$request->getOrdersCount()" as a valid number >=1

Reply:
I have already checked it at the first _if_ block of the function.